subprojects/packagefiles/nanobind/meson.build

```diff
@@ -0,0 +1,91 @@
+project(
+    'nanobind',
+    'cpp',
+    version: '2.0.0',
+    meson_version: '>=0.53.0',
+    license: 'BSD-3-Clause',
+)
+
+py_mod = import('python')
+py = py_mod.find_installation()
+py_dep = py.dependency(embed: true)
+
+robin_map_dep = dependency('robin-map')
+
+incdir = include_directories('include')
+
+# Arguments are derived from nanobind_build_library function in CMake configuration
+dep_compile_args = []
+dep_link_args = []
+if get_option('python.allow_limited_api')
+  py_version = py.language_version()
+
+  if py_version.version_compare('>=3.12')
+    add_project_arguments('-DPy_LIMITED_API=0x030C0000', language: 'cpp')
+  endif
+endif
+
+if get_option('buildtype') == 'shared'
+  if host_machine.system() == 'darwin'
+    is_pypy = py_dep.get_variable('implementation_lower') == 'pypy'
+    if is_pypy
+      resp_file = meson.project_source_root() / 'cmake/darwin-ld-pypy.sym'
+    else
+      resp_file = meson.project_source_root() / 'cmake/darwin-ld-cpython.sym'
+    endif
+    add_project_link_arguments('-Wl,-dead_strip', '-Wl,x', '-Wl,-S', '-Wl,@' + resp_file)
+    dep_link_args += ['-Wl,-dead_strip', '-Wl,x', '-Wl,-S', '-Wl,@' + resp_file]
+  elif host_machine.system() != 'windows'
+    add_project_link_arguments('-Wl,-s', language: 'cpp')
+    dep_link_args += ['-Wl,-s']
+  endif
+
+  add_project_arguments('-DNB_BUILD', language: 'cpp')
+  dep_compile_args += ['-DNB_SHARED']
+elif host_machine.system() != 'windows' and host_machine.system() != 'darwin'
+  dep_compile_args += ['-ffunction-sections', '-fdata-sections']
+  dep_link_args += ['-Wl,--gc-sections']
+endif
+
+compiler = meson.get_compiler('cpp')
+if compiler.get_id() == 'msvc'
+  dep_compile_args += ['-D_CRT_SECURE_NO_WARNINGS']
+else
+  # Discussion in WrapDB PR #1556 as to whether or not this is necessary
+  # add_project_arguments('-fno-strict-aliasing', language: 'cpp')
+endif
+
+# Here the CMake configuration links against Python::SABIModule on Windows
+# if building an abi3 target - does Meson handle this internally?
```

Comment:
When a project uses:

```
py.extension_module(
    'foomod',
    'foomod.c',
    limited_api: '3.12',
)
```
meson links to python3.dll rather than python312.dll. I assume that's what "SABIModule" is.

Reply:
Seems reasonable. I assume it would be more idiomatic to remove the code that defines Py_LIMITED_API in this configuration and just let Meson handle that